### ISSUE TITLE
Webpack externals don't seem to be excluded in webpack5

### ISSUE THREAD

**Issue by vaunus** [lostpebble/generate-package-json-webpack-plugin#35]
We've recently updated to webpack 5 and I noticed that some externals, in particular `aws-sdk` are still being added to the package.json generated by this plugin at build time. We're using a monorepo which the root of which is referred to below by `workspaceRoot`.

The relevant parts of my webpack config are as follows:
```
. . .
      externalsPresets: { node: true },
      externals: [
        nodeExternals({
          modulesDir: path.resolve(workspaceRoot, 'node_modules')
        }),
        'aws-sdk',
      ],
      plugins: [
        . . .
        new GeneratePackageJsonPlugin(
          {
            engines: {
              node: `~14`
            },
            description: entryKey,
            license: 'UNLICENSED',
            repository: {}
          },
          {
            useInstalledVersions: true,
            resolveContextPaths: [`${workspaceRoot}`],
            sourcePackageFilenames: [`${workspaceRoot}/package.json`]
          }
        )
      ]
. . . 
```

I then get this package.json:
```
{
  "engines": {
    "node": "~14"
  },
  "license": "UNLICENSED",
  "repository": {},
  "dependencies": {
    "aws-sdk": "2.1008.0",
    . . .
  }
}
```

I was expecting `aws-sdk` to not be in here. Perhaps I am misunderstanding something?

**Comment by lostpebble** [comment]
Hi @vaunus ,

Externals that are used by your code are always included in the `package.json` file. Are you sure your bundled code does not reference `aws-sdk` anywhere? Otherwise yes, it would be strange that it is appearing in the final output package.json.

**Comment by vaunus** [comment]
Right I understand, yes we are using `aws-sdk` in our code so that explains it!

Is there any way of explicitly excluding a package from the generated `package.json` file?

The use case for us is probably quite a common one. We are building for AWS Lambda which already has `aws-sdk` runtimes built in. Given it is ~70M in node_modules I would rather not install it.

**Comment by lostpebble** [comment]
Ah... okay, I see. That is an interesting use case- though I would think AWS would just not install it then? Since its a native dependency. I would have thought that you deploy your bundled code with the `package.json` (as opposed to an entire installed `node_modules` folder) and they take care of the installation stuff on their side. But I'm not clued up on it.

I think it shouldn't be too hard to add an option to forcefully exclude found dependencies. I'll look into it soon.

**Comment by lostpebble** [comment]
@vaunus I just released `2.5.0` please use this version, try out the new option:

```ts
{
  excludeDependencies: ["aws-sdk"],
}
```

**Comment by vaunus** [comment]
Thanks so much for the quick turnaround @lostpebble. I've just tested this and it's working great! 🙂
